manifests/interface.pp

```diff
@@ -739,13 +749,25 @@
     }
 
     'RedHat': {
-      file { "/etc/sysconfig/network-scripts/ifcfg-${name}":
-        ensure  => $ensure,
+      if versioncmp($::operatingsystemmajrelease, '8') >= 0 {
+        service { 'NetworkManager':
```

Comment:
This should not stay in the interface define, but in the main network class. Also, as it is now, this will bring a duplicated resource error when 2 interfaces are declared (an  if ! defined(Service['NetworkManager'] would be needed).

Reply:
I see. This makes sense. I will change this tomorrow and resolve when done.

Reply:
I used my approach with your hint. If you totally not agree with it just tell me. In the end I just my code merged somehow. I don't want to maintain a fork